File: app/src/main/java/com/frogobox/cleaner/service/AlarmBoosterBroadcastReceiver.java
package com.frogobox.cleaner.service;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;

import com.frogobox.cleaner.view.fragment.PhoneBoosterFragment;

import com.frogobox.cleaner.myapplication.R;

/**
 * Created by Frogobox Software Industries 3/2/2017.
 */

public final class AlarmBoosterBroadcastReceiver extends BroadcastReceiver {
    SharedPreferences.Editor editor;
    SharedPreferences sharedpreferences;
    @Override
    public void onReceive(Context context, Intent intent) {

        sharedpreferences = context.getSharedPreferences("waseem", Context.MODE_PRIVATE);
//        Toast.makeText(context, "Alarm worked.", Toast.LENGTH_LONG).show();



        /// when memory is orveloaded or increased


        editor = sharedpreferences.edit();
        editor.putString("booster", "1");
        editor.commit();

        try {
            PhoneBoosterFragment.optimizebutton.setBackgroundResource(0);
            PhoneBoosterFragment.optimizebutton.setImageResource(0);
            PhoneBoosterFragment.optimizebutton.setImageResource(R.drawable.optimize);
        }
        catch(Exception e)
        {

        }

    }
}

File: app/src/main/java/com/frogobox/cleaner/service/AlarmBoosterBroadcastReceiver.kt
package com.frogobox.cleaner.service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.frogobox.cleaner.R
import com.frogobox.cleaner.utils.Constant
import com.frogobox.cleaner.view.fragment.ChargeBoosterFragment

/**
 * Created by Frogobox Software Industries 3/2/2017.
 */

class AlarmBoosterBroadcastReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        logicOnReceive(context, intent)
    }

    private fun logicOnReceive(context: Context, intent: Intent) {

        val sharedpreferences = context.getSharedPreferences(Constant.Variable.SHARED_PREF_WASEEM, Context.MODE_PRIVATE)
        val editor = sharedpreferences.edit()
        editor.putString(Constant.Variable.SHARED_PREF_BOOSTER, "1")
        editor.commit()

        try {
            ChargeBoosterFragment.optimizebutton.setBackgroundResource(0)
            ChargeBoosterFragment.optimizebutton.setImageResource(0)
            ChargeBoosterFragment.optimizebutton.setImageResource(R.drawable.bg_button_optimize)
        } catch (e: Exception) {

        }

    }

}
